lib/locater.js

```diff
@@ -7,52 +7,56 @@
 function findIndices(target, line) {
   var indices = [];
 
-  if (typeof target === 'string') {
-    var index = 0;
-    var offset = 0;
+  //converts all targets to regex
+  if (!(target instanceof RegExp)) {
+    target = new RegExp('\\b' + target + '\\b', "g");
+  }
 
-    while ((index = line.indexOf(target, offset)) !== -1) {
-      indices.push(index);
-      offset = index + 1;
-    }
-  } else if (target instanceof RegExp) {
-    while ((match =  target.exec(line)) !== null) {
-      target.lastIndex = match.index + 1;
-      indices.push(match.index);
-    }
+  while ((match =  target.exec(line)) !== null) {
+    indices.push(match.index);
   }
 
   return indices;
 }
 
+/**
+ * Removes the last line break from a string.
+ * @param line {String} - the string to remove line break from
+ */
+function removeLastLineBreaks(source) {
+  return source.replace(/[\n]+$/, '')
+}
+
+/**
+ * Converts sting to lines based on \n. Also removes last line break
+ * @param line {String} - the string split
+ */
+function splitIntoLines(source) {
+  return source.split('\n');
+}
+
 module.exports = {
   find: function (target, source) {
-    var sourceLines = source.split('\n');
-    var lineNumber = 1;
+    var sourceLines = splitIntoLines( removeLastLineBreaks(source) );
     var results = [];
 
-    sourceLines.forEach(function (line, idx) {
-      // Do not count the last line that is empty
-      if (idx === sourceLines.length - 1 && ! line) {
-        return;
-      }
-
-      var indices = findIndices(target, line);
-
-      indices.forEach(function (index) {
-        var position = {line: lineNumber, cursor: index + 1};
+    if (!(target instanceof Array)) target = [target] //convert all targets to array
 
-        results.push(position);
-      });
+    sourceLines.forEach(function (line, lineNumber) {
+      target.forEach(function(currentValue) {
+        var indices = findIndices(currentValue, line);
 
-      lineNumber++;
+        indices.forEach(function (index) {
+          results.push( { line: lineNumber + 1, cursor: index + 1 } );
```

Comment:
Why do we have `{ line: lineNumber + 1, cursor: index + 1 }` compared to `{ line: lineNumber, cursor: index + 1 }` in the old version?